drivers/serial/uart_ns16550.c

```diff
@@ -681,7 +681,7 @@ static void uart_ns16550_irq_tx_enable(const struct device *dev)
 		 * different states.
 		 */
 		for (uint8_t i = 0U; i < num_cpu_states; i++) {
-			pm_constraint_set(cpu_states[i].state);
+			pm_policy_state_disallow(cpu_states[i].state);
```

Comment:
how they can restrict all states? PM_STATE_ACTIVE is at zero, so what state will be allowed to be in if all are not allowed?

Reply:
`PM_STATE_ACTIVE` is effectively ignored (policy will return `NULL`, indicating CPU must remain active).

Reply:
Would it be the same if disallow will come when other state is set?
Adding short info in function api doc would be useful.
e.g.
```
Request to disallow all states will result in CPU staying in active state, until restrictin is lifted.
```

Reply:
clarified